rust/connlib/tunnel/src/tests/reference.rs

```diff
@@ -189,40 +184,80 @@ impl ReferenceState {
                 10,
                 state.client.inner().ipv4_cidr_resource_dsts(),
                 |ip4_resources| {
-                    icmp_to_cidr_resource(
-                        packet_source_v4(state.client.inner().tunnel_ip4),
-                        sample::select(ip4_resources).prop_flat_map(crate::proptest::host_v4),
-                    )
+                    prop_oneof![
+                        icmp_to_destination(
+                            packet_source_v4(state.client.inner().tunnel_ip4),
+                            select_host_v4(ip4_resources.clone())
+                        ),
```

Comment:
```suggestion
                        icmp_packet(
                            packet_source_v4(state.client.inner().tunnel_ip4),
                            select_host_v4(&ip4_resources)
                        ),
```

Will this fit onto one line if we:

- Extract `tunnel_ip4` into a variable further up
- Make `select_host_v4` accept a reference
- Rename the function